### DISCUSSION TITLE
Getting µCNC master up and running - problems

### DISCUSSION THREAD

**Discussion by hmmueller** [Paciente8159/uCNC#711] [Q&A] [answered]
(No more other threads - I promise ...)
Ok, current state with µCNC on VSCode+PlatformIO:
- Uploaded
- Motors get warm, drivers get hot, screws are kept tightly by the motors = (motor) currents are flowing.
- µCNC replies to a few commands on Serial Monitor:

````
...
20:06:26.269 > $132=200.000
20:06:26.275 > ok
G00 X10
20:06:31.939 > ok
?20:07:17.861 > <Hold:0|MPos:0.000,0.000,0.000|FS:0.000,0|Pn:HZ>
$
20:08:04.087 > [HLP:$$ $# $G $I $N $x=val $Nx=line $J=line $C $X $H ~ ! ? ctrl-x]
20:08:04.092 > ok
$#
20:08:14.060 > error:1
$$
````

Some of the lines are red, whatever that means. But after a few commands, it 
a) locks up; e.g. the final $$ above never returned an answer.
b) and the Z motor starts "double ticking" (first with a longer interval, then about every second) ... sounds like a driver is getting so hot that it is crying for help? (at this time, the heat sinks on the 3 DRV8825A are burning hot).
c) Last, the G00 X10 (I expected some move for 10 mm) did not move anything. Seems something is still locked/disabled.

(Edit: The error:1 after <span>$</span># is mine. I'm quite sure <span>$</span># means "a <span>$</span>...=... command"; so error:1 is just a normal syntax error).

I am not sure I know what I should try now for a) or b) or c). Any help appreciated!

Thanks!
H.M.

**Comment by Paciente8159** [comment, marked as answer]
a) I disabled control pins and uploaded the firmware to an UNO board and had no problems

```
20:39:15.121 > Grbl 1.10 [uCNC v1.10.beta '$' for help]
20:39:15.124 > >:ok
20:39:15.124 > >:ok
20:39:15.193 > ok
20:39:15.896 > ok
20:39:16.080 > ok
$
20:39:23.039 > [HLP:$$ $# $G $I $N $x=val $Nx=line $J=line $C $X $H ~ ! ? ctrl-x]
20:39:23.044 > ok
$#
20:39:31.914 > [G54:0.000,0.000,0.000]
20:39:31.914 > [G55:0.000,0.000,0.000]
20:39:31.918 > [G56:0.000,0.000,0.000]
20:39:31.921 > [G57:0.000,0.000,0.000]
20:39:31.923 > [G58:0.000,0.000,0.000]
20:39:31.923 > [G59:0.000,0.000,0.000]
20:39:31.925 > [G28:0.000,0.000,0.000]
20:39:31.929 > [G30:0.000,0.000,0.000]
20:39:31.931 > [G92:0.000,0.000,0.000]
20:39:31.934 > [TLO:0.000]
20:39:31.934 > [PRB:0.000,0.000,0.000:0]
20:39:31.936 > ok
$$
20:39:37.243 > $0=33.333
20:39:37.243 > $1=0
20:39:37.243 > $2=0
20:39:37.243 > $3=0
20:39:37.243 > $4=0
20:39:37.245 > $5=0
20:39:37.245 > $6=0
20:39:37.245 > $7=0
20:39:37.245 > $10=1
20:39:37.245 > $11=0.200
20:39:37.248 > $12=0.002
20:39:37.248 > $13=0
20:39:37.248 > $20=0
20:39:37.251 > $21=0
20:39:37.251 > $22=0
20:39:37.251 > $23=0
20:39:37.251 > $24=10.000
20:39:37.253 > $25=50.000
20:39:37.253 > $26=250
20:39:37.253 > $27=2.000
20:39:37.256 > $30=1000
20:39:37.256 > $31=0
20:39:37.256 > $32=0
20:39:37.256 > $81=0.000
20:39:37.259 > $100=200.000
20:39:37.259 > $101=200.000
20:39:37.262 > $102=200.000
20:39:37.262 > $110=500.000
20:39:37.262 > $111=500.000
20:39:37.264 > $112=500.000
20:39:37.264 > $120=10.000
20:39:37.267 > $121=10.000
20:39:37.267 > $122=10.000
20:39:37.267 > $130=200.000
20:39:37.270 > $131=200.000
20:39:37.270 > $132=200.000
20:39:37.272 > ok
G0X10
20:39:58.669 > ok
?20:39:59.906 > <Run|MPos:6.635,0.000,0.000|FS:490.964,0|Pn:XYZ|WCO:0.000,0.000,0.000>
?20:40:00.056 > <Run|MPos:7.745,0.000,0.000|FS:400.602,0|Pn:XYZ|Ov:100,100,100>
?20:40:00.175 > <Run|MPos:8.480,0.000,0.000|FS:328.313,0|Pn:XYZ>
?20:40:00.316 > <Run|MPos:9.160,0.000,0.000|FS:243.976,0|Pn:XYZ>
?20:40:00.463 > <Run|MPos:9.655,0.000,0.000|FS:159.639,0|Pn:XYZ>
?20:40:00.933 > <Idle|MPos:10.000,0.000,0.000|FS:0.000,0|Pn:XYZ>
?20:40:01.171 > <Idle|MPos:10.000,0.000,0.000|FS:0.000,0|Pn:XYZ>
?20:40:01.773 > <Idle|MPos:10.000,0.000,0.000|FS:0.000,0|Pn:XYZ>
```

All is running

b) That is an hardware problem. If the driver is getting too hot you need to tune the current on those DRV8825A. [See this video](https://www.youtube.com/watch?v=BV-ouxhZamI)

c) That is normal. Your hold pin is active. The status message shows that 
`?20:07:17.861 > <Hold:0|MPos:0.000,0.000,0.000|FS:0.000,0|Pn:HZ>`
Status says hold and inputs for pins HOLD and LIMIT_Z are active
Sending any motion command will get on hold. Also note that wile on hold your commands will start to get hold in line, in the serial buffer. Once you reach the limit of 128 chars you will get and overflow on the buffer and all commands will be discarded with an associated error message.

**Reply by hmmueller** [reply]
Thank you! 
a) All is running here too.
b) Changed the max current to1.6A (probably would have worked before; but more heat = shorter lifespan);
c) I jumpered the HOLD pin, and now everything works. BUT: If I say ?, I get the reply

<Idle|MPos:0.000,0.000,0.000|FS:0.000,0|Pn:Z|WCO:0.000,0.000,0.000>

So that would mean the Z limit switch is enabled. Two reasons why I doubt that: (1) It is simply jumpered on the CNCShield, like the X and Y limit switch connectors; and (2) the machine moves!! So what does this Pn:Z actually mean??

H.M.

**Comment by Paciente8159** [comment]
It means that the Z limit is active.
It will only take effect if you enable hard limits. Otherwise limits will be ignored.

If Z is active either you configured the wrong pin, or you have a setting to invert the pin logic, or you have a damaged pin.

**Reply by hmmueller** [reply]
Ok - that was easy (slowly, I learn ;-)): I took the AVR-UNO environment - which has the Z limit switch at pin 4. However, with CNCShield, I should have taken (and now take) AVR-CNC-Shield-V3, which redefines #define LIMIT_Z_BIT 3. Uploaded - and Pn:Z is gone.

Thanks a lot for your super-quick help!

Tomorrow I hope to replace the jumpers on the limit switches with the real switches - let's see whether they work as expected.

Good night!
H.M.

**Reply by hmmueller** [reply]
... and all end switches also do their duty. So I herewith declare this thread to be solved and closed, because of utter success. Thank you again!

H.M.
